rl/agents/sarsa.py

```diff
@@ -0,0 +1,196 @@
+from rl.core import Agent
+import warnings
+from copy import deepcopy
+
+import numpy as np
+from keras.callbacks import History
+
+from rl.callbacks import TestLogger, TrainEpisodeLogger, TrainIntervalLogger, Visualizer, CallbackList
+from rl.agents.dqn import mean_q
+from rl.util import huber_loss
+from keras.layers import Input, Lambda
+from keras.models import Model
+import keras.backend as K
+from rl.policy import EpsGreedyQPolicy
+from rl.util import get_object_config
+
+
+class Sarsa(Agent):
+    def __init__(self, model, nb_actions, policy=EpsGreedyQPolicy(), gamma=.99, nb_steps_warmup=10,
```

Comment:
Important: Set `policy=None` in the constructor and then set it to the default value in the body of the init method if none:
```
if policy is None:
    policy = EpsGreedyQPolicy()
```
Otherwise different instances will share the same policy instance, which leads to all sorts of problems. It's a really stupid Python behavior. You can refer to the DQNAgent to see how to do it properly (I fixed it there recently).